workflow/validate/validate.go

```diff
@@ -1424,9 +1424,11 @@ func validateDAGTaskArgumentDependency(arguments wfv1.Arguments, ancestry []stri
 
 	for _, param := range arguments.Parameters {
 		if param.Value == nil {
-			return errors.Errorf(errors.CodeBadRequest, "missing value for parameter '%s'", param.Name)
-		}
-		if strings.HasPrefix(param.Value.String(), "{{tasks.") {
+			// Disallow ValueFrom.Parameter, but allow ValueFrom.ConfigMapKeyRef
+			if param.ValueFrom == nil || param.ValueFrom.Parameter != "" {
+				return errors.Errorf(errors.CodeBadRequest, "missing value for parameter '%s'", param.Name)
+			}
```

Comment:
Nice fix. I see that `ValueFrom` has a lot of sub-parameter options. How did you decide which to allow and which not to?

Reply:
I just wanted to concentrate on the description in the ticket and allow as little as possible. Further, I wanted to align the behaviour of steps and dag for this very example.

Could be that other options might be valid here as well, though. That should be proposed and discussed in a follow up ticket.

Reply:
I just wanted to concentrate on the description in the ticket and allow as little as possible. Further, I wanted to align the behaviour of steps and dag for this very example.

Could be that other options might be valid here as well, though. That should be proposed and discussed in a follow up ticket.

Reply:
allow as little as possible? But I think it will allow all of the other members to be set, right? (just `ValueFrom.Parameter` is prevented)

Reply:
You are completely right.